src/util/timer.rs

```diff
@@ -86,9 +88,29 @@ impl<T> Ord for TimeoutTask<T> {
     }
 }
 
+lazy_static! {
+    pub static ref GLOBAL_TIMER_HANDLE: Handle = start_global_timer();
```

Comment:
How about use `tokio::runtime` and `tokio::timer::Delay` to avoid the thread? there is a little test, and we can see `Delay` won't block the current thread even if we poll it on `runtime::current_thread`:
```rust
#![allow(unused_imports)]
extern crate futures;
extern crate tokio;
use futures::{future, stream, Future, Stream};
use std::thread;
use std::time::{Duration, Instant};
use tokio::executor::thread_pool;
use tokio::runtime::{self, current_thread};
use tokio::timer::{Delay, Interval};

fn main() {
    let i1 = Interval::new(Instant::now(), Duration::from_secs(5))
        .for_each(|_| {
            println!("tick on interval 1");
            future::ok::<_, _>(())
        })
        .map_err(|_| ());

    let i2 = Delay::new(Instant::now() + Duration::from_secs(8))
        .map(|_| println!("delay 2"))
        .map_err(|_| ());

    let mut runtime = current_thread::Runtime::new().unwrap();
    runtime.spawn(i1).spawn(i2).run().unwrap();
}
```
And we can see there is only 1 thread in the process with `top -H -p <pid>`.

Reply:
How to integrate it with `CpuPool`?

Reply:
can it become a bottleneck if we all use the global handler?

Reply:
It's possible, but it's easy to start a new timer.

Reply:
`Delay` and `Internal` are futures. We can use them in CpuPool directly.

Reply:
Have you verified this? I believe they can only be polled inside `Runtime` using this way.